frontend/controller/actions/group.js

```diff
@@ -295,16 +292,9 @@ export default (sbp('sbp/selectors/register', {
 
       // When syncing the group contract, the contract might call /remove on
       // itself if we had previously joined and left the group. By using
-      // deferredRemove we ensure that it's not deleted until we've finished
+      // ephemeral we ensure that it's not deleted until we've finished
       // trying to join.
-      // When we are re-joinining, this function will call /cancelRemove to
-      // ensure that the contract isn't removed, and if we're not re-joining,
-      // the call at the end to /remove with removeIfPending will remove the
-      // group contract.
-      // Part of this same functionality is what `sbp('okTurtles.data/set', 'JOINING_GROUP-'`
-      // does. In a later improvement, we could remove that and handle re-joinining
-      // within the contract state.
-      await sbp('chelonia/contract/sync', params.contractID, { deferredRemove: true })
+      await sbp('chelonia/contract/retain', params.contractID, { ephemeral: true })
```

Comment:
OK... but where do we call `retain` in a persistent way for joining the group?

Don't we need to do that somewhere? And if not, why not?

(Also a request for an explanation regarding this in the code itself via comments)